src/kibana/plugins/dashboard/index.js

```diff
@@ -70,182 +71,171 @@ define(function (require) {
     .directive('dashboardApp', function (Notifier, courier, AppState,
                                             timefilter, kbnUrl, searchAuditor) {
         return {
-          link: function(scope, elem) {
-              elem.on('submit', function() {
-                  // TODO: find what event kibana4 uses for searches to listen instead
-                  scope.state.query =
-                      {
-                          query_string : {
-                              query : searchAuditor.autoCapQuery(scope.state.query['query_string']['query'])
-                          }
-                      };
-                  scope.filterResults();
-                  searchAuditor.logQuery(scope.state.query['query_string']['query'],
-                                         datemath.parse(scope.timefilter.time.from),
-                                         datemath.parse(scope.timefilter.time.to));
-              });
-          },
-        controller: function ($scope, $route, $routeParams, $location, $http, 
-           configFile, Private, getAppState, saveRuleManager, indexPatterns) {
-        indexPatterns.refresh();
-        $scope.saveRuleManager = saveRuleManager;
-        var queryFilter = Private(require('components/filter_bar/query_filter'));
+          controller: function ($scope, $route, $routeParams, $location, $http, 
+             configFile, Private, getAppState, saveRuleManager, indexPatterns) {
+            indexPatterns.refresh();
+            $scope.saveRuleManager = saveRuleManager;
+            var queryFilter = Private(require('components/filter_bar/query_filter'));
 
-        var notify = new Notifier({
-          location: 'Dashboard'
-        });
+            var notify = new Notifier({
```

Comment:
here I just removed the link function cause it's no longer needed, that functionality was moved to the searchAuditor service and to $scope.filterResults.  The rest of the changes is messed up formatting from my editor which I will push up a fix for